src/Simplex/Messaging/Agent/Store/SQLite.hs

```diff
@@ -686,3 +730,31 @@ updateHashSnd_ dbConn connAlias SndMsgData {..} =
       ":conn_alias" := connAlias,
       ":last_internal_snd_msg_id" := internalSndId
     ]
+
+-- * Broadcast helpers
+
+checkBroadcast :: DB.Connection -> BroadcastId -> IO (Either StoreError a) -> IO (Either StoreError a)
+checkBroadcast dbConn bId action =
+  withTransaction dbConn $ do
+    ok <- bcastExists_ dbConn bId
+    if ok then action else pure $ Left SEBcastNotFound
```

Comment:
could we remove callback from the helper, please? - this can be moved to calling functions, it just reads better if the logic flow is flattened there

Reply:
or maybe it's okay.. not sure, kind of a grew a bit on me after several functions